api/envoy/config/filter/network/tcp_proxy/v2/tcp_proxy.proto

```diff
@@ -131,4 +140,22 @@ message TcpProxy {
   // The maximum number of unsuccessful connection attempts that will be made before
   // giving up. If the parameter is not specified, 1 connection attempt will be made.
   google.protobuf.UInt32Value max_connect_attempts = 7 [(validate.rules).uint32.gte = 1];
+
+  // Allows for specification of multiple upstream clusters along with weights
+  // that indicate the percentage of traffic to be forwarded to each cluster.
+  // The router selects an upstream cluster based on these weights.
+  message WeightedCluster {
+    message ClusterWeight {
+      // Name of the upstream cluster.
+      string name = 1 [(validate.rules).string.min_bytes = 1];
+
+      // When a request matches the route, the choice of an upstream cluster is
+      // determined by its weight. The sum of weights across all entries in the
+      // clusters array determines the total weight.
+      google.protobuf.UInt32Value weight = 2 [(validate.rules).uint32.gte = 1];
```

Comment:
uint32, not g.p.UInt32Value

Reply:
Addressed in 9c55555.

P.S. I'm amazed at the number of fives in the hash. :)